### DISCUSSION TITLE
cannot get Server side pagination and sorting to work properly on q-table

### DISCUSSION THREAD

**Discussion by michalakis** [quasarframework/quasar#12170] [General - Components / Directives / etc]
I'm not sure if this is the right place to post this. If it isnt, I apologize, here is my problem:

Filter works just fine

sorting 
-stays stuck on descending: false after I click on a row to change the sorting. 
-The onRequest method detects that a column was clicked, and registers the column, but the sorting doesnt change 

pagination: 
-when records per page is changed, the request event fires, and has the correct rows per page value, but clicking on the page number does not fire the request event 
- also, records per page does not change to reflect the new value, it stays stuck on the first value

here is the code for the q-table

```
    <q-table
      class="multiple-select-table z-0"
      :title="tableTitle"
      :rows="rows"
      :columns="columns"
      row-key="key"
      :selected-rows-label="getSelectedString"
      selection="multiple"
      v-model:selected="selected"
      v-model:pagination="pagination"
      ref="table"
      :filter="filter"
      :rows-per-page-options="[10,50,100]"
      @request="onRequest"
      binary-state-sort
    >
```

and the pagination:
```
<template v-slot:pagination><!--="props-->
  <div class="flex flex-center">
    <span class="q-mx-lg">{{getPageItemNumbers(pagination.page, pagination.rowsPerPage, pagination.rowsNumber)}}</span>
    <q-pagination
      class="table-pagination"
      color="theme-3"
      v-model="pagination.page"
      :max-pages="7"
      :max="pagination.pagesNumber"
      :boundary-numbers="true"
      direction-links
      icon-prev="fas fa-chevron-left"
      icon-next="fas fa-chevron-right"
      :to-fn="pageFunction"
    />
  </div>
</template>
```

**Comment by code-is-art** [comment]
[This worked for me.](https://github.com/quasarframework/quasar/discussions/13735)
